setup.py

```diff
@@ -535,15 +536,24 @@ def pxd(name):
         'depends': _pxi_dep['sparse']},
     '_libs.tslib': {
         'pyxfile': '_libs/tslib',
-        'pxdfiles': ['_libs/src/util'],
+        'pxdfiles': ['_libs/src/util',
+                     '_libs/tslibs/conversion',
+                     '_libs/tslibs/timedeltas',
+                     '_libs/tslibs/timezones',
+                     '_libs/tslibs/nattype'],
         'depends': tseries_depends,
```

Comment:
Does it matter that `np_datetime` is included in 'depends' via `tseries_depends` but not in 'pxdfiles'?

Putting e.g. conversion.pxd into tslib's 'pxdfiles' entry adds it to the `sources` kwarg of Extension.  Why is that the desired outcome?  (and uh, what does that do?  the distutils [docs](https://docs.python.org/3/distutils/apiref.html#distutils.core.Extension) don't actually say anything useful)

Reply:
the disutils docs are pretty bad.

I went with the premise that if we are cimporting then we care if the api changes.  to test is easy, change something in nattype.pxd, you get recompiles in lots of things, but not everything (e.g. fields/frequencies and such). change timedeltas.pxd and you get less.

this has the desired behavior of only dependency based compilation.

I think the sources/depends are actually much more on the c-level and are separate from this. (but I don't think this is really an issue as we basically include _libs/src in the compilation each time so that the c-headers are there).